src/plugins/intel_cpu/src/nodes/interaction.cpp

```diff
@@ -2,27 +2,157 @@
+
+        Xbyak::Label move_scale_loop_label;
+        Xbyak::Label move_scale_end_label;
+
+        if (jcp_.with_scales && jcp_.broadcast_scales) {
+            uni_vmovss(Xmm(vmm_scales.getIdx()), ptr[reg_scales]);
+            uni_vbroadcastss(vmm_scales, Xmm(vmm_scales.getIdx()));
+        }
+
+        mov(reg_in_aux, reg_in);
+        mov(reg_out_aux, reg_out);
+        if (jcp_.with_scales && !jcp_.broadcast_scales) {
+            mov(reg_scales, ptr[reg_params + GET_OFF(p_scales)]);
+        }
+        size_t tail_size = jcp_.work_amount % vec_size;
+        L(move_scale_loop_label);
+        {
+            cmp(reg_work_amount, vec_size);
+            jl(move_scale_end_label, T_NEAR);
+
+            load_scale_store(vec_size);
+
+            sub(reg_work_amount, vec_size);
+
+            jmp(move_scale_loop_label, T_NEAR);
+        }
+        L(move_scale_end_label);
+        if (tail_size) {
+            load_scale_store(tail_size);
+        }
+
+        this->postamble();
+
+        for (const auto& emitter : emitters) {
+            if (emitter.second)
+                emitter.second->emit_data();
+        }
+    }
+
+    void load_scale_store(size_t step) {
+        bool is_tail = step < vec_size;
+
+        load(vmm_in, reg_in_aux, jcp_.src_prc, step, is_tail);
+
+        if (jcp_.with_scales) {
+            if (!jcp_.broadcast_scales) {
+                load(vmm_scales, reg_scales, Precision::FP32, step, is_tail);
+                add(reg_scales,  sizeof(float) * step);
+            }
+            uni_vmulps(vmm_in, vmm_in, vmm_scales);
+        }
+
+        store(reg_out_aux, vmm_in, jcp_.dst_prc, step);
+
+        if (!is_tail) {
+            add(reg_in_aux, jcp_.src_prc.size() * step);
+            add(reg_out_aux, jcp_.dst_prc.size() * step);
+        }
+    }
+#undef GET_OFF
+
+    inline void load(const Vmm& vmm_dst, const Xbyak::Reg64& reg_src, Precision src_prc, const int& elt_num, bool fill) {
+        const auto seed = load_emitter_params(src_prc, Precision::FP32, elt_num, fill, "float_min").hash();
+        if (!emitters[seed]) {
+            emitters[seed].reset(new jit_load_emitter(this, isa, src_prc, Precision::FP32, elt_num, Precision::FP32, fill, "float_min"));
+        }
+
+        emitters[seed]->emit_code({static_cast<size_t>(reg_src.getIdx()), 0}, {static_cast<size_t>(vmm_dst.getIdx())},
+                                  pool_aux_vmm_idxs, pool_aux_gpr_idxs);
+    }
+    inline void store(const Xbyak::Reg64& reg_dst, const Vmm& vmm_src, Precision dst_prc, const int& elt_num) {
+        const auto seed = store_emitter_params(Precision::FP32, dst_prc, elt_num).hash();
+        if (!emitters[seed]) {
+            emitters[seed].reset(new jit_store_emitter(this, isa, Precision::FP32, dst_prc, elt_num));
+        }
+
+        emitters[seed]->emit_code({static_cast<size_t>(vmm_src.getIdx()), 0}, {static_cast<size_t>(reg_dst.getIdx())},
+                                  pool_aux_vmm_idxs, pool_aux_gpr_idxs);
+    }
```

Comment:
if do not have scales, we can keep precision when load and store without precision convert to fp32. It's just data movement if I understand correctly.

Reply:
Yes you are right.